### PR TITLE
Backport PR 11983 to beta

### PR DESCRIPTION
**Pull request by ef4**
Per request by @rwjblue.

### CODE REVIEW

File: packages/ember-metal/lib/computed.js [hunk 1]
```diff
@@ -390,49 +403,72 @@ ComputedPropertyPrototype.get = function(obj, keyName) {
...
-  var setter         = this._setter;
-  var meta           = metaFor(obj, cacheable);
-  var cache          = meta.cache;
-  var hadCachedValue = false;
-
-  var cachedValue, ret;
-
-  if (this._readOnly) {
-    throw new EmberError(`Cannot set read-only property "${keyName}" on object: ${inspect(obj)}`);
+  // cache requires own meta
```
Comment (stefanpenner) [inline packages/ember-metal/lib/computed.js]: doesn't this take a different path if volatile?

**Comment by rwjblue** [comment]
Thank you!

**Comment by denzo** [comment]
Also, "set" does not return the value of the object itself anymore. Instead it returns the value being set. Thus chaining is no longer possible. Was that done intentionally?

**Comment by rwjblue** [comment]
@denzo - Yes, that was intentional. See related issue https://github.com/emberjs/ember.js/issues/12095.

**Pull request merged**